### PR TITLE
implements a tcp repeater, with tests

### PR DESCRIPTION
**Pull request by mcfunley**
This adds test coverage for the (existing) udp repeater, and adds a tcp version. 

The only config change is that `repeaterProtocol` can now be `"tcp"`.

### CODE REVIEW

File: test/repeater_tests.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    this.servers.start(function() { 
+      repeater.send('foobar');
+    });
+  },
+
+  tcp_repeater_works: function(test) {
+    test.expect(1);
+
+    var statsd = new TcpStatsD(9125, function(packet, rinfo) {
+      test.equal('foobar\n', packet.toString());
```
Comment (mcfunley) [inline test/repeater_tests.js]: I noticed while writing this test that the tcp server emits events with newlines, and the udp server doesn't. I'm not sure if this matters. Happy to make them uniform in a separate PR if we think it does.

Comment (benburry) [inline test/repeater_tests.js]: @mcfunley I think that's fine - newlines seem to have become the defacto way to indicate the end of a packet in the statsd persistent connection world

**Comment by benburry** [comment]
Thanks for the contribution!
